pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/JUnitTestsShouldIncludeAssertRule.java

```diff
@@ -189,4 +203,29 @@ private boolean isExpectStatement(ASTStatementExpression expression,
         }
         return false;
     }
+
+    private boolean isSoftAssertionStatement(ASTStatementExpression expression,
+                                             Map<String, VariableNameDeclaration> variables) {
+        if (expression != null) {
+            ASTPrimaryExpression pe = expression.getFirstChildOfType(ASTPrimaryExpression.class);
+            if (pe != null) {
+                Node name = pe.getFirstDescendantOfType(ASTName.class);
+                if (name != null) {
+                    String img = name.getImage();
+                    if (img.indexOf(".") == -1) {
+                        return false;
+                    }
+                    String[] tokens = img.split("\\.");
+                    String methodName = tokens[1];
+                    boolean methodIsAssertAll = "assertAll".equals(methodName);
+
+                    String varName = tokens[0];
+                    boolean variableTypeIsSoftAssertion = variables.containsKey(varName) && variables.get(varName).getTypeImage().endsWith("SoftAssertions");
```

Comment:
We probably should use `net.sourceforge.pmd.lang.java.typeresolution.TypeHelper.isA(TypeNode, String)` for the type check, e.g. `TypeHelper.isA(variables.get(varName), "org.assertj.core.api.SoftAssertions")`.
Note: we need to add a method in TypeHelper, that deals with a TypeNameDeclaration and a class given as String.